frontend/src/pages/customer/Customers.js

```diff
@@ -0,0 +1,198 @@
+import React, {useEffect, useState} from 'react';
+import Table from "react-bootstrap/Table";
+import Form from 'react-bootstrap/Form'
+import {FaEdit} from "react-icons/fa";
+import Page from "../../components/Page";
+import Button from "react-bootstrap/Button";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import TogglePage from "../../components/TogglePage";
+import ModalAddCustomer from "./ModalAddCustomer";
+import ModalEditCustomer from "./ModalEditCustomer";
+import CardContainer from "../../components/CardContainer";
+import ErrorMessage from "../../messages/errorMessage";
+
+export default () => {
+
+    const [page, setPage] = useState({
+        active: 1,
+        currentPage: 1,
+        countPerPage: 10,
+        countPages: 1
+    });
+    const [customers, setCustomers] = useState([]);
+    const [filter, setFilter] = useState([]);
+    const [lgShow, setLgShow] = useState(false);
+    const [editCustomer, setEditCustomer] = useState({
+        editShow: false,
+        customer: []
+    });
+    const [errorMessage, setErrors] = useState('');
+    const filterOptions={'All': '', 'Active': true, 'Disabled': false};
+
+    const onChangeFilter = (e) => {
+        e.preventDefault();
+        setFilter(e.target.value);
+        getCustomers('/customers?status=' + e.target.value + '&size=' + page.countPerPage);
+
+    };
+
+    const handleCountPerPage = (e) => {
+        e.preventDefault();
+        setPage(preState => ({
+            ...preState,
+            countPerPage: e.target.value
+        }));
+        getCustomers('/customers?size=' + e.target.value + '&status=' + filter);
+    };
+
+    const handleChangeStatus = (e) => {
+        let status = e.target.value !== 'true';
+        let id = e.target.id;
+        fetch('/customers/' + id + '/status', {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(status)
+        })
+            .then(function (response) {
+                if (response.status !== 200) {
+                    setErrors("Something go wrong, try later");
+                } else {
+                    let newData = [...customers];
+                    newData.forEach(elem => {
+                        if (elem.id == id) {
+                            elem.active = status;
+                        }
+                    });
+                    setCustomers(newData);
+                }
+            });
+    };
+
+
+    const changePage = (e) => {
+        e.preventDefault();
+        let page = e.target.innerHTML - 1;
+        getCustomers('/customers?page=' + page + '&status=' + filter);
+    };
+
+    useEffect(() => {
+        getCustomers('/customers');
+    }, []);
+
+
+    function getCustomers(url) {
+        fetch(url)
```

Comment:
as an alternative you could use async/await approach.
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/async_function